### PR TITLE
ci: migrate to fastify reusable workflow

### PR DESCRIPTION
**Pull request by Fdawgs**
#### Checklist

- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
CI fails :(

**Comment by Fdawgs** [comment]
Requires #76

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,45 +1,15 @@
...
-    needs: test
-    runs-on: ubuntu-latest
-    permissions:
-      pull-requests: write
-      contents: write
-    steps:
-      - uses: fastify/github-action-merge-dependabot@v3
-        with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v3
```
Comment (Eomm) [inline .github/workflows/ci.yml:15]: Could we add the linting too?

Comment (Fdawgs) [reply .github/workflows/ci.yml:15]: Linting is already part of this module's `test` script

Comment (Eomm) [reply .github/workflows/ci.yml:15]: I mean to split the package.json script to run it once

**Pull request merged**